### ISSUE TITLE
[HOLD for payment 2024-07-22] [$250] Categorizing- Workspace chat under To field opens not here page when new WS is not created yet

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#42552]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.75-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to FAB > Track expense
3. Track a manual expense
4. In 1:1 DM, click Categorize it from the whisper
5. Click New workspace
6. Select a category
7. On confirmation page, click on the workspace chat under To field

## Expected Result:
Since the workspace chat is not created yet, the workspace chat should be disabled (not clickable)

## Actual Result:
The workspace chat is clickable and leads to not here page

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/5b0d2831-5fd2-4043-b42e-6a006695dc52

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c1df32a642cf9a1e</li>
        <li>Upwork Job ID: 1794118174987796480</li>
        <li>Last Price Increase: 2024-05-31</li>
<li>Automatic offers: </li>
<ul>
<li>rojiphil | Reviewer | 102573818</li>
<li>tienifr | Contributor | 102573820</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mallenexpensify</details>

**Comment by lanitochka17** [comment]
@mallenexpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by mallenexpensify** [comment]
Was able to reproduce. _Seems_ like it could be external. @rojiphil , if you agree, can you review the proposals above? Thx


https://github.com/Expensify/App/assets/22508304/53cafcc8-9516-403b-b630-a9abdcd49c84

**Comment by rojiphil** [comment]
@mallenexpensify @Expensify/Design
I need a reconfirmation on the expected behavior mentioned in the OP as I am not sure if this is intentional. 
To me, it looks like we should have created the new workspace and moved ahead instead of restricting the user by disabling(not clickable) workspace chat. This way the user could proceed like any other existing workspace. 
What do you think?

**Comment by dannymcclain** [comment]
>To me, it looks like we should have created the new workspace and moved ahead instead

Would love some more thoughts because I wasn't as involved in these convos, but I think you're right about this. To me it seems like we should have already created the workspace by the time they get to the confirmation screen—so clicking on that row would behave just like any other existing workspace. @shawnborton does that sound like where the Slack convos landed?

**Comment by shawnborton** [comment]
Not entirely sure, but I kind of think we wouldn't create the workspace until the expense is successfully sent there? I don't feel strongly though. 

If we do opt for something like that, then I agree, I would just make that workspace row not pressable/hoverable (but **please** do not make it look disabled, because it's not a disabled UI component, its just read-only text)

**Comment by dubielzyk-expensify** [comment]
> I would just make that workspace row not pressable/hoverable (but please do not make it look disabled, because it's not a disabled UI component, its just read-only text)

That feels right to me as well

**Comment by rojiphil** [comment]
Thanks for confirming the expected behaviour here. 
So the consensus is not to create a new workspace until expense is successfully requested. 

@tienifr [proposal](https://github.com/Expensify/App/issues/42552#issuecomment-2130906454) LGTM as the proposal not only solves the problem but also addresses the need of not looking like disabled. However, the final UI changes also require the involvement of the design team but this can be taken up in the PR stage. 
🎀👀🎀 C+ reviewed

**Comment by rojiphil** [comment]
Not overdue. Awaiting PR

**Comment by mallenexpensify** [comment]
@tienifr as [mentioned in CONTRIBUTING.md](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#:~:text=Contributors%20should%20not%20submit%20proposals%20on%20issues%20when%20they%20have%20assigned%20issues%20or%20PRs%20that%20are%20awaiting%20an%20action%20from%20them.%20If%20so%2C%20they%20will%20be%20in%20violation%20of%20Rule%20%231%20(Get%20Shit%20Done)%20in%20our%20Code%20of%20Conduct%20and%20will%20receive%20a%20warning.%20Multiple%20warnings%20can%20lead%20to%20removal%20from%20the%20program.)
> Contributors should not submit proposals on issues when they have assigned issues or PRs that are awaiting an action from them. If so, they will be in violation of Rule #1 (Get Shit Done) in our [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) and will receive a warning. Multiple warnings can lead to removal from the program.

Please focus on submitting a PR for this job you're hired for vs [posting new proposals. ](https://github.com/Expensify/App/issues/43064#issuecomment-2149020754) Thx.

**Comment by hayata-suenaga** [comment]
@tienifr [this blocker](https://github.com/Expensify/App/pull/43153#issuecomment-2151163090) right?

putting hold on this issue 😄

**Comment by mallenexpensify** [comment]
Thanks for the details and for putting on hold @tienifr and @hayata-suenaga

**Comment by hayata-suenaga** [comment]
The proposals are still being tested in the issue that is holding this one -> https://github.com/Expensify/App/issues/43073#issuecomment-2166916532

**Comment by mallenexpensify** [comment]
Thanks @tienifr , taken off hold.  
@rojiphil , looks like the PR is waiting on you
- https://github.com/Expensify/App/pull/43153#issuecomment-2194349047

**Comment by mallenexpensify** [comment]
@rojiphil is the PR awaiting your review again?

**Comment by rojiphil** [comment]
> is the PR awaiting your review again?

Yes. Will review today.

**Comment by hayata-suenaga** [comment]
Melvin, The PR is on production now

**Comment by rojiphil** [comment]
* [x]  [@rojiphil] The PR that introduced the bug has been identified. Link to the PR: [Offending PR](https://github.com/Expensify/App/pull/23802)
* [x]  [@rojiphil] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: [Added comment](https://github.com/Expensify/App/commit/c5cfe64787dd66131cac177ae073f45876717538#r144615286)
* [x]  [@rojiphil] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: Not Required. Existing checklist is good enough to capture such issues.
* [x]  [@rojiphil] Determine if we should create a regression test for this bug. :  Since we, now, use interactive pattern here (i.e. shown normal cursor instead of disabled cursor), it is better to include regression tests 
* [x]  [@rojiphil] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again. 

Test Steps
	1	Create a track expense
	2	Perform the action `Categorize it`
	3	Press New workspace
	4	Select a category
	5	Verify the workspace chat under To field is not pressable and shows normal cursor while hovering over

**Comment by mallenexpensify** [comment]
Contributor: @tienifr due $250 via NewDot  
Contributor+: @rojiphil due $250 via Upwork. 

@rojiphil , can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~01125570305f069950 

Test Case
- https://github.com/Expensify/Expensify/issues/415745

**Comment by rojiphil** [comment]
> can you please accept the job and reply here once you have?

@mallenexpensify Offer accepted. Thanks

**Comment by mallenexpensify** [comment]
@rojiphil paid! Updated payment comment above.

**Comment by JmillsExpensify** [comment]
$250 approved for @tienifr
